core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -234,116 +190,86 @@ public FileSystem getFileSystem() {
-      Objects.requireNonNull(getFileSystem());
-      Objects.requireNonNull(getConfiguration());
-      Objects.requireNonNull(getTableConfiguration());
+    public AccumuloConfiguration getTableConfiguration() {
+      return tableConfiguration;
     }
-  }
 
-  /** Builder interface parallel to {@link FileAccessOperation}. */
-  protected interface FileAccessOperationBuilder<SubbuilderType> extends NeedsFile<SubbuilderType>,
-      NeedsFileSystem<SubbuilderType>, NeedsTableConfiguration<SubbuilderType> {
-    // no optional/generic methods.
-  }
-
-  /**
-   * Operation object for performing {@code getFileSize()} operations.
-   */
-  protected class GetFileSizeOperation extends FileAccessOperation<GetFileSizeOperation>
-      implements GetFileSizeOperationBuilder {
-    /** Return the size of the file. */
-    @Override
-    public long execute() throws IOException {
-      validate();
-      return getFileSize(this);
+    public RateLimiter getRateLimiter() {
+      return rateLimiter;
     }
-  }
-
-  /**
-   * Builder interface for {@link GetFileSizeOperation}, allowing execution of {@code getFileSize()}
-   * operations.
-   */
-  public interface GetFileSizeOperationBuilder
-      extends FileAccessOperationBuilder<GetFileSizeOperationBuilder> {
-    /** Return the size of the file. */
-    long execute() throws IOException;
-  }
 
-  /**
-   * Options common to all {@code FileOperation}s which perform reading or writing.
-   */
-  protected static class FileIOOperation<SubclassType extends FileIOOperation<SubclassType>>
-      extends FileAccessOperation<SubclassType> {
-    private RateLimiter rateLimiter;
-
-    /** Specify a rate limiter for this operation. */
-    @SuppressWarnings("unchecked")
-    public SubclassType withRateLimiter(RateLimiter rateLimiter) {
+    public FileStuff rateLimiter(RateLimiter rateLimiter) {
       this.rateLimiter = rateLimiter;
-      return (SubclassType) this;
+      return this;
     }
 
-    public RateLimiter getRateLimiter() {
-      return rateLimiter;
+    /** Check for null parameters. */
+    public void validate() {
+      Objects.requireNonNull(getFilename());
+      Objects.requireNonNull(getFileSystem());
+      Objects.requireNonNull(getConfiguration());
+      Objects.requireNonNull(getTableConfiguration());
     }
   }
 
-  /** Builder interface parallel to {@link FileIOOperation}. */
-  protected interface FileIOOperationBuilder<SubbuilderType>
-      extends FileAccessOperationBuilder<SubbuilderType> {
-    /** Specify a rate limiter for this operation. */
-    SubbuilderType withRateLimiter(RateLimiter rateLimiter);
-  }
-
   /**
    * Operation object for constructing a writer.
    */
-  protected class OpenWriterOperation extends FileIOOperation<OpenWriterOperation>
-      implements OpenWriterOperationBuilder, NeedsFileOrOuputStream<OpenWriterOperationBuilder> {
+  public class WriterBuilder extends FileStuff implements WriterTableConfiguration {
```

Comment:
Since this extends FileStuff it may introduce invalid Fluent paths.

Reply:
I can try making FileStuff's methods protected.